modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_4_4/test/DDMStructureUpgradeProcessTest.java

```diff
@@ -0,0 +1,237 @@
+					"createDate, modifiedDate, parentStructureId, classNameId,",
+					"structureKey, version, name, description, definition, ",
+					"storageType, type_, lastPublishDate ) values ( ?, ?, ?, ",
+					"?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ",
+					"?);"))) {
+
+			java.util.Date utilDate = new java.util.Date();
+
+			java.sql.Date sqlDate = new java.sql.Date(utilDate.getTime());
+
+			preparedStatement.setLong(1, RandomTestUtil.randomLong());
+			preparedStatement.setLong(2, RandomTestUtil.randomLong());
+			preparedStatement.setString(3, RandomTestUtil.randomString());
+			preparedStatement.setString(4, RandomTestUtil.randomString());
+			preparedStatement.setLong(5, structure.getStructureId());
+			preparedStatement.setLong(6, RandomTestUtil.randomLong());
+			preparedStatement.setLong(7, RandomTestUtil.randomLong());
+			preparedStatement.setLong(8, RandomTestUtil.randomLong());
+			preparedStatement.setString(9, RandomTestUtil.randomString());
+			preparedStatement.setLong(10, RandomTestUtil.randomLong());
+			preparedStatement.setString(11, RandomTestUtil.randomString());
+			preparedStatement.setDate(12, sqlDate);
+			preparedStatement.setDate(13, sqlDate);
+			preparedStatement.setLong(14, RandomTestUtil.randomLong());
+			preparedStatement.setLong(
+				15, PortalUtil.getClassNameId(DDMFormInstance.class.getName()));
+			preparedStatement.setString(16, RandomTestUtil.randomString());
+			preparedStatement.setString(17, RandomTestUtil.randomString());
+			preparedStatement.setString(18, structure.getName());
+			preparedStatement.setString(19, RandomTestUtil.randomString());
+			preparedStatement.setString(20, structure.getDefinition());
+			preparedStatement.setString(21, RandomTestUtil.randomString());
+			preparedStatement.setInt(22, RandomTestUtil.randomInt());
+			preparedStatement.setDate(23, sqlDate);
+
+			preparedStatement.executeUpdate();
+		}
+
+		try (Connection connection = DataAccess.getConnection();
+			PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"insert into DDMStructureVersion (",
+					"structureVersionId, structureId, ctCollectionId, ",
+					"definition) values ( ?, ?, ?, ?);"))) {
+
+			preparedStatement.setLong(1, RandomTestUtil.randomLong());
+			preparedStatement.setLong(2, structure.getStructureId());
+			preparedStatement.setLong(3, RandomTestUtil.randomLong());
+			preparedStatement.setString(4, structure.getDefinition());
+
+			preparedStatement.executeUpdate();
+		}
+
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				_CLASS_NAME, LoggerTestUtil.WARN)) {
+
+			UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+				_upgradeStepRegistrator, _CLASS_NAME);
+
+			upgradeProcess.upgrade();
+
+			List<LogEntry> logEntries = logCapture.getLogEntries();
+
+			if (!logEntries.isEmpty()) {
+				LogEntry logEntry = logEntries.get(0);
+
+				for (LogEntry entry : logEntries) {
```

Comment:
Are you sure we need to loop through `logEntries` in this test since we only have one DDMStructure entry?
Please check other tests that use logCapture